ios/MaasCore/Sources/MaasCore/Api/Utils/RefreshToken+Publishers.swift

```diff
@@ -41,8 +41,13 @@ extension Publisher where Failure == ApiError {
         retryWithRecovery(
             recoveryPublisher: Maas.apiConfig.refreshTokenPublisher(),
             failureShouldRetry: {
-                if case .unauthorized = $0 { return true }
-                else { return false }
+                switch $0 {
+                case .unauthorized: return true
+                case .errorCode(let code, _):
+                    return code == .user(.tokeninvalidorexpired) ||
+                        code == .user(.refreshtokeninvalidorexpired)
```

Comment:
if we handle specific error codes tokeninvalidorexpired || refreshtokeninvalidorexpired should we still trigger token refresh on `401` - (.unauthorized)?

Reply:
Seems like yes, based on yesterday's test where an error code is not returned in case the token is empty 🙂

Reply:
I don't think we can handle `refreshtokeninvalidorexpired` in the SDK as we don't own the refresh token.

Reply:
yesterday's test was flawed 😅 there i used a fake token corrupt, so the token wasn't actually expired.. today my token got old, and i received `tokeninvalidorexpired`

Reply:
But if SDK users provide an invalid (or empty, or nil) id token out of the box, should we still ask for a new token and retry?

Reply:
probably not, I think we should retry only when we get explicit error code, that token is expired 🤔  do you agree?

Reply:
OK 🙂